src/test/java/org/littil/api/guestTeacher/repository/GuestGuestTeacherRepositoryTest.java

```diff
@@ -1,52 +1,74 @@
 import static org.mockito.Mockito.doReturn;
-import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.when;
 
 @QuarkusTest
-@Disabled("Disabled, needs a lot of refactoring")
 class GuestGuestTeacherRepositoryTest {
 
     @InjectSpy
     GuestTeacherRepository repository;
 
     @Test
     void givenFindExistingTeacherByName_thenShouldReturnSuccessfully() {
+        String firstName = RandomStringUtils.randomAlphabetic(10);
         String surname = RandomStringUtils.randomAlphabetic(10);
-        List<GuestTeacherEntity> teacher = List.of(new GuestTeacherEntity(UUID.randomUUID(), surname, null, null, null, null));
+        List<GuestTeacherEntity> teacherList = List.of(new GuestTeacherEntity(UUID.randomUUID(), firstName, surname, null, null, null, null));
 
-        PanacheQuery<GuestTeacherEntity> query = mock(PanacheQuery.class);
-        doReturn(query).when(repository).find("surname", surname);
-        when(query.list()).thenReturn(teacher);
+        doReturn(teacherList).when(repository).list("surname like ?1", "%" + surname + "%");
 
         List<GuestTeacherEntity> foundTeacher = repository.findBySurnameLike(surname);
 
-        assertThat(teacher).isEqualTo(foundTeacher);
+        assertThat(teacherList).isEqualTo(foundTeacher);
     }
 
     @Test
     void givenFindNonExistingTeacherByName_thenShouldReturnEmptyOptional() {
         final String searchSurname = RandomStringUtils.randomAlphabetic(10);
 
-        PanacheQuery<GuestTeacherEntity> query = mock(PanacheQuery.class);
-        doReturn(query).when(repository).find("surname", searchSurname);
-        when(query.list()).thenReturn(Collections.emptyList());
+        doReturn(Collections.emptyList()).when(repository).list("surname like ?1", "%" + searchSurname + "%");
 
         List<GuestTeacherEntity> foundTeacher = repository.findBySurnameLike(searchSurname);
 
         assertThat(foundTeacher).isEmpty();
     }
+
+    @Test
+    void givenFindFilledExistingTeacherByName_thenShouldReturnSuccessfully() {
+        String firstName = RandomStringUtils.randomAlphabetic(10);
+        String surname = RandomStringUtils.randomAlphabetic(10);
+        String prefix = RandomStringUtils.randomAlphabetic(10);
+        LocationEntity location = new LocationEntity();
+        LinkedHashSet<DayOfWeek> availibility = new LinkedHashSet<>(List.of(DayOfWeek.MONDAY));
+        UserEntity user = new UserEntity();
+        GuestTeacherEntity expectedTeacher = new GuestTeacherEntity(UUID.randomUUID(), firstName, surname, prefix, location, availibility, user);
+
+        doReturn(List.of(expectedTeacher)).when(repository).list("surname like ?1", "%" + surname + "%");
+
+        List<GuestTeacherEntity> foundTeacherList = repository.findBySurnameLike(surname);
+
+        assertThat(foundTeacherList).isNotEmpty();
+
+        GuestTeacherEntity foundTeacher = foundTeacherList.get(0);
+
+        assertThat(expectedTeacher.getFirstName()).isEqualTo(foundTeacher.getFirstName());
```

Comment:
This is correct, but you should checkout https://joel-costigliola.github.io/assertj/assertj-core-features-highlight.html#field-by-field. That's a cool feature that makes you write less code ;). Let me know if you want to implement this or just go with what you got now.

Reply:
Nice tip! Although these methods are deprecated now. It's replaced by usingRecursiveComparison().